### PR TITLE
Ignore empty where clauses when building SQL

### PR DESCRIPTION
**Pull request by nolamesa**

### CODE REVIEW

**Comment by nolamesa** [comment]
This allows us to build where statements conditionally.

```python 
where = ""
if foo:
   where = "foo"
...
select('x').from_('tbl').where(where).where(conditions.get('bar'))
```

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -913,6 +913,9 @@ def where(
         Returns:
             Select: the modified expression.
         """
+        expressions = [exp for exp in expressions if exp is not None and exp != ""]
```
Comment (barakalon) [inline sqlglot/expressions.py:916]: I think this should work the same:

```python
expressions = [exp for exp in expressions if exp]
```

It's a little less explicit, but I don't think there are other arguments we're expecting that could evaluate to False?

Comment (barakalon) [reply sqlglot/expressions.py:916]: Also, do we want the same thing for `having`? Maybe we should move this into `_apply_conjuction_builder`?

Comment (nolamesa) [reply sqlglot/expressions.py:916]: yeah you are right but see my test, I am guessing `where(False)` is valid

Comment (tobymao) [reply sqlglot/expressions.py:916]: in his test he allows where(False) which is different, although i wonder if this is correct

Comment (tobymao) [reply sqlglot/expressions.py:916]: why would you ever do where False, i think this is tricky, and probably not necessary

Comment (nolamesa) [reply sqlglot/expressions.py:916]: it is correct, tried it in postgres

Comment (nolamesa) [reply sqlglot/expressions.py:916]: you are right I can move into the conjunction builder, having should do the same

**Review by barakalon — APPROVED** [review]

**Pull request merged**